templates/redis.conf.epp

```diff
@@ -173,7 +245,7 @@ rdbcompression <% if @rdbcompression -%>yes<% else -%>no<% end %>
 rdbchecksum yes
 
 # The filename where to dump the DB
-<% if @dbfilename %>dbfilename <%= @dbfilename %><% else %># dbfilename dump.rdb<% end %>
+<% if $dbfilename { %>dbfilename <%= $dbfilename %><% } else { %># dbfilename dump.rdb<% } %>
```

Comment:
How about we fix this a layer higher and in config.pp do `pick($dbfilename, 'dump.rdb')`. Then require a `String[1]` as a value.

Reply:
I guess the Point here is to decide, if this Line should be active or not.  It's not so much the Filename itself.  

In `manifests/instance.pp` we could do something like
```
  $_real_dbfilename = if $dbfilename =~ String {
    $dbfilename
  } elsif $dbfilename = true {
    'dump.rdb'
  } else {
    undef
  }
```

But that would not help that much for the Template to decide between active or inactive Line.

Reply:
This would also help, that if the User passes the Boolean `true` the resulting Filename would also be `true`.

Reply:
Oh, my apologies; I completely missed the comment bit in `else`. Now it makes sense to have `if/else`.